### PR TITLE
Add guided attention loss for Tacotron2

### PR DESCRIPTION
**Pull request by kan-bayashi**
Should be merged after #815

### CODE REVIEW

**Comment by kan-bayashi** [comment]
Ready to merge.

File: espnet/nets/pytorch_backend/e2e_tts_tacotron2.py [hunk 1]
```diff
@@ -43,6 +43,108 @@ def make_non_pad_mask(lengths):
     return seq_range_expand < seq_length_expand
 
 
+class GuidedAttentionLoss(torch.nn.Module):
+    """Guided attention loss function
```
Comment (r9y9) [inline espnet/nets/pytorch_backend/e2e_tts_tacotron2.py]: Better to add a paper reference?

**Comment by sw005320** [comment]
- can we use this for ASR? if so, it's better to design so that we can use this for both ASR and TTS
- can you add a specific test for this guided attention?

**Comment by sw005320** [comment]
Correction. You seem to add some test. So please ignore my suggestion for the test part.

**Comment by kan-bayashi** [comment]
> can we use this for ASR? if so, it's better to design so that we can use this for both ASR and TTS

It can be used for ASR.
What we need to calculate GuidedAttentionLoss are ilens, olens, att_ws.
https://github.com/espnet/espnet/blob/3f79e79639d09885cb1678e57dcd7d41d4399b86/espnet/nets/pytorch_backend/e2e_tts_tacotron2.py#L68-L75

But I do not want to support this loss for ASR in this PR.

**Comment by sw005320** [comment]
OK

File: espnet/nets/pytorch_backend/e2e_tts_tacotron2.py (hunk 1 above)
Comment (kan-bayashi) [reply espnet/nets/pytorch_backend/e2e_tts_tacotron2.py]: Added! Thank you for your review.
